### ISSUE TITLE
Build fails with -DNO_MATHOMS

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8687]
Migrated from [rt.perl.org#40963](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=40963) (status was 'resolved')

Searchable as RT40963$

**Comment by p5pRT** [comment]
### From @jdhedden

#### Created by jhedden@sundev3.bloomberg.com



Adding the following argument to 'Configure'&#8203;:

&nbsp;    \-A append&#8203;:ccflags='\-DNO\_MATHOMS'

causes the build to fail&#8203;:

&nbsp;          CCCMD =  /opt/SUNWspro/bin/cc \-DPERL\_CORE \-c
\-DPERL\_PATCHNUM=29343 \-D\_REENTRANT \-DNO\_MATHOMS \-DDEBUGGING
\-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64  \-g  
/opt/SUNWspro/bin/cc \-L/usr/lib \-L/usr/ccs/lib
\-L/opt/SUNWspro/WS6U1/lib \-L/usr/local/lib  \-o miniperl \\
&nbsp;            \`echo   gv\.o toke\.o perly\.o op\.o pad\.o regcomp\.o dump\.o
util\.o mg\.o reentr\.o hv\.o av\.o perl\.o run\.o pp\_hot\.o sv\.o pp\.o scope\.o
pp\_ctl\.o pp\_sys\.o doop\.o doio\.o regexec\.o utf8\.o taint\.o deb\.o
universal\.o xsutils\.o globals\.o perlio\.o perlapi\.o numeric\.o mathoms\.o
locale\.o pp\_pack\.o pp\_sort\.o  | sed 's/ op\.o / /'\` \\
&nbsp;            miniperlmain\.o opmini\.o \-lsocket \-lnsl \-ldl \-lm \-lpthread
\-lc 
Undefined                       first referenced
 symbol                             in file
Perl\_save\_I8                        sv\.o
ld&#8203;: fatal&#8203;: Symbol referencing errors\. No output written to miniperl
make&#8203;: \*\*\* \[miniperl\] Error 1


This further indicates that 'Configure \-Dno\_mathoms' is non\-functional\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.9.5:

Configured by jhedden at Fri Nov 17 08:47:41 EST 2006.

Summary of my perl5 (revision 5 version 9 subversion 5 patch 29343)
configuration:
  Platform:
    osname=solaris, osvers=2.9,
archname=sun4-solaris-thread-multi-64int
    uname='sunos sundev3 5.9 generic_112233-12 sun4u sparc
sunw,sun-fire '
    config_args='-de -Dusethreads=yes -Duse64bitint=yes -Dusedevel=yes
-Doptimize=-g -DDEBUGGING -DPERL_TRACK_MEMPOOL'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define,
usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='/opt/SUNWspro/bin/cc', ccflags ='-D_REENTRANT -DDEBUGGING
-I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-g',
    cppflags='-D_REENTRANT -DDEBUGGING -I/usr/local/include'
    ccversion='Sun WorkShop 6 update 1 C 5.2 2000/09/11',
gccversion='', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=87654321
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long long', ivsize=8, nvtype='double', nvsize=8,
Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='/opt/SUNWspro/bin/cc', ldflags =' -L/usr/lib -L/usr/ccs/lib
-L/opt/SUNWspro/WS6U1/lib -L/usr/local/lib '
    libpth=/usr/lib /usr/ccs/lib /opt/SUNWspro/WS6U1/lib /usr/local/lib
    libs=-lsocket -lnsl -lgdbm -ldl -lm -lpthread -lc
    perllibs=-lsocket -lnsl -ldl -lm -lpthread -lc
    libc=/lib/libc.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-KPIC', lddlflags='-G -L/usr/lib -L/usr/ccs/lib
-L/opt/SUNWspro/WS6U1/lib -L/usr/local/lib'


Characteristics of this binary (from libperl): 
  Compile-time options: DEBUGGING MULTIPLICITY PERL_DONT_CREATE_GVSV
                        PERL_IMPLICIT_CONTEXT PERL_MALLOC_WRAP
                        PERL_TRACK_MEMPOOL USE_64_BIT_INT USE_ITHREADS
                        USE_LARGE_FILES USE_PERLIO USE_REENTRANT_API
  Locally applied patches:
        DEVEL
  Built under solaris
  Compiled at Nov 20 2006 10:10:57
  %ENV:
    PERLIO="perlio"
  @INC:
    lib
    /usr/local/lib/perl5/5.9.5/sun4-solaris-thread-multi-64int
    /usr/local/lib/perl5/5.9.5
   
/usr/local/lib/perl5/site_perl/5.9.5/sun4-solaris-thread-multi-64int
    /usr/local/lib/perl5/site_perl/5.9.5
    .


Environment for perl 5.9.5:
    HOME=/home/jhedden
    LANG=C
    LANGUAGE=C
    LC_ALL=C
   
LD_LIBRARY_PATH=/bb/bin/informix/ifmx7.3/lib:/bb/bin/informix/ifmx7.3/lib/esql:/bb/bin/informix/ifmx7.3/lib/cli:/bb/bin/informix/ifmx7.3/lib/tools:/home/jhedden/local/lib:/usr/local/lib:/bb/data/sunlibs/5::/opt/centera/Centera_SDK/lib/32
    LOGDIR (unset)
   
PATH=:/bb/bin/informix/ifmx7.3/bin:/bb/bin/informix/bin:/home/jhedden/local/bin:/bbsrc/tools/bbcm:/bbsrc/tools/bbcm/parent:/usr/atria/bin:/bb/data/local/java/bin:/bb/data/local/ant/bin:/bb/data/local/bin:/bb/bin/cvsacl/bin:/usr/local/bin:/usr/bin:/usr/bin/X11:/usr/openwin/bin:/bb/shared/bin:/bb/shared/abin:/bb/bin/robo:/bb/bin:/usr/sbin:/usr/ccs/bin:/opt/SUNWspro/bin:/usr/ucb:/opt/ins++/bin.solaris:.
    PERLIO=perlio
    PERL_BADLANG (unset)
    SHELL=/usr/bin/ksh



 
____________________________________________________________________________________
Sponsored Link

Mortgage rates near 39yr lows. $510k for $1,698/mo. 
Calculate new payment! www.LowerMyBills.com/lre

```
</details>

**Comment by p5pRT** [comment]
### From @jdhedden

This also fails under Perl 5\.8\.8\.


 
\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
Sponsored Link

Mortgage rates near 39yr lows\. $420k for $1\,399/mo\. 
Calculate new payment\! 
www\.LowerMyBills\.com/lre

**Comment by p5pRT** [comment]
### From @jdhedden

> This also fails under Perl 5\.8\.8\.

Correction&#8203;: Fails under the patch 28220 snapshot \(i\.e\.\, 
perl\-28220\.tar\.bz2\) for Perl 5\.8\.8\.  \(mathoms\.c is not in the base 5\.8\.8
distribution\.\)

**Comment by p5pRT** [comment]
@jdhedden - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @rgs

That's fixed by change \#29476\.

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'resolved'
